WooCommerce/Classes/Extensions/Date+StartAndEnd.swift

```diff
@@ -12,10 +12,25 @@ extension Date {
     /// Returns self's end of day in the given time zone.
     func endOfDay(timezone: TimeZone) -> Date {
         let calendar = createCalendar(timezone: timezone)
-        var components = DateComponents()
-        components.day = 1
-        components.second = -1
-        return calendar.date(byAdding: components, to: startOfDay(timezone: timezone))!
+
+        let startOfNextDay: Date = {
+            var components = DateComponents()
+            components.day = 1
+            guard let nextDay = calendar.date(byAdding: components, to: startOfDay(timezone: timezone)) else {
+                fatalError("The next day cannot be calculated for \(self) with time zone \(timezone)")
```

Comment:
Do you think we should use `logErrorAndExit` so we can monitor the occurrence of these?

Reply:
Great point, I forgot that the error message in `fatalError` isn't logged in Sentry. Updated in https://github.com/woocommerce/woocommerce-ios/pull/7704/commits/f921d01c11e4268295b63b94b1017ea4af0f1764